components/renku_data_services/crc/blueprints.py

```diff
@@ -328,24 +335,56 @@ async def _patch(
             class_id: int,
             user: base_models.APIUser,
         ):
-            return await self._put_patch(user, resource_pool_id, class_id, body)
+            res = await self.repo.update_resource_class(
+                user, resource_pool_id, class_id, put=False, **body.model_dump(exclude_none=True)
+            )
+            return json(apispec.ResourceClassWithId.model_validate(res).model_dump(exclude_none=True))
 
         return "/resource_pools/<resource_pool_id>/classes/<class_id>", ["PATCH"], _patch
 
-    async def _put_patch(
-        self,
-        api_user: base_models.APIUser,
-        resource_pool_id: int,
-        class_id: int,
-        body: apispec.ResourceClassPatch | apispec.ResourceClass,
-    ):
-        cls = await self.repo.update_resource_class(
-            api_user=api_user,
-            resource_pool_id=resource_pool_id,
-            resource_class_id=class_id,
-            **body.model_dump(exclude_none=True),
-        )
-        return json(apispec.ResourceClassWithId.model_validate(cls).model_dump(exclude_none=True))
+    def get_tolerations(self) -> BlueprintFactoryResponse:
+        """Get all tolerations of a resource class."""
+
+        @authenticate(self.authenticator)
+        @only_admins
+        async def _get_tolerations(_: Request, resource_pool_id: int, class_id: int, user: base_models.APIUser):
+            res = await self.repo.get_tolerations(user, resource_pool_id, class_id)
+            return json([i for i in res])
+
+        return "/resource_pools/<resource_pool_id>/classes/<class_id>/tolerations", ["GET"], _get_tolerations
+
+    def delete_tolerations(self) -> BlueprintFactoryResponse:
+        """Delete all tolerations of a resource class."""
+
+        @authenticate(self.authenticator)
+        @only_admins
+        async def _delete_tolerations(_: Request, resource_pool_id: int, class_id: int, user: base_models.APIUser):
+            await self.repo.delete_tolerations(user, resource_pool_id, class_id)
+            return HTTPResponse(status=204)
```

Comment:
use `sanic.empty(status=204)` instead as that's what it's made for. instead of the HTTPResponse base class